web3.js/src/transaction.ts

```diff
@@ -172,8 +191,22 @@ export class Transaction {
   /**
    * Construct an empty Transaction
    */
-  constructor(opts?: TransactionCtorFields) {
-    opts && Object.assign(this, opts);
+  constructor(opts?: TransactionCtorFields | TransactionCtorFields_DEPRECATED) {
+    if (!opts) {
+      return;
+    } else if (Object.prototype.hasOwnProperty.call(opts, 'recentBlockhash')) {
+      console.warn(
+        'Creating a Transaction with `recentBlockhash` is deprecated. ',
+      );
+      const oldOpts = opts as TransactionCtorFields_DEPRECATED;
+      Object.assign(this, oldOpts);
+      this.recentBlockhash = oldOpts.recentBlockhash;
```

Comment:
The first line implies the second line, doesn't it?

Reply:
You know, I can't remember now why we had this set up like this. I think this was a leftover from something else we were trying and didn't remove. I'll update this!